sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/SQLKeywordUtils.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+import scala.util.Try
+
+import org.apache.spark.sql.catalyst.parser.{CatalystSqlParser, SqlBaseLexer}
+
+private[sql] object SQLKeywordUtils {
```

Comment:
can we follow `SQLKeywordSuite` and read the keywords list from a file?

Reply:
How about making SQLKeywordSuite follow what we do here?

`SqlBaseLexer.VOCABULARY` is a stable public API and widely used

Reply:
FYI, https://github.com/search?q=Lexer.VOCABULARY&type=code

Reply:
My worry is how to determine reserved keywords. Invoking the parser for each keyword looks a bit overkill to me.

Reply:
I know. Before implementing as a TVF, it looks overkilling to me too. But now it does not stay in the Optimizer seems OK to me. FYI, I have found that [trino](https://github.com/trinodb/trino/blob/master/core/trino-parser/src/main/java/io/trino/sql/ReservedIdentifiers.java#L144) also implements it in this way.

Reply:
It is not static, but relies on ANSI mode and `spark.sql.ansi.enforceReservedKeywords`

Reply:
And I think it's not very necessary keep the map in memory. For JDBC tools, it may only use once while creating the connection. e.g. https://github.com/apache/hive/blob/ba0217ff17501fb849d8999e808d37579db7b4f1/beeline/src/java/org/apache/hive/beeline/SQLCompleter.java#L56

Reply:
if it's only used once by the thriftserver, why do we bother to add a TVF? We can't assume that a public SQL function may only be called once.

Reply:
addressed, please take another look